python/res/job_queue/job_manager.py

```diff
@@ -467,15 +469,19 @@ def extract_stderr_stdout(self, job):
                     extra_fields.update({"stdout": stdout})
         return extra_fields
 
-    def exit(self, job, exit_status, error_msg):
-        self.dump_EXIT_file(job, error_msg)
+    def markJobFailure(self, job, exit_status, error_msg):
```

Comment:
Would it make sense to merge `markJobFailure` and `markJobSuccess` and instead call it `markJob` and pass in a boolean `success` or have it be conditional on the exit status?

Reply:
Or even put this as internal logic and have `markJobComplete` call it? If one is always expected to call `a()` and then `b()` I would say that the API is a bit off...

Reply:
Could not agree more!